src/mbgl/renderer/tile_pyramid.cpp

```diff
@@ -215,12 +215,14 @@ void TilePyramid::update(const std::vector<Immutable<style::LayerProperties>>& l
         }
     }
 
-    if (type != SourceType::Annotations) {
+    if (type != SourceType::Annotations && cacheEnabled) {
         auto conservativeCacheSize = static_cast<size_t>(
             std::max(static_cast<double>(parameters.transformState.getSize().width) / tileSize, 1.0) *
             std::max(static_cast<double>(parameters.transformState.getSize().height) / tileSize, 1.0) *
             (parameters.transformState.getMaxZoom() - parameters.transformState.getMinZoom() + 1) * 0.5);
         cache.setSize(conservativeCacheSize);
+    } else {
```

Comment:
Should this be `else if (!cacheEnabled)`. Otherwise the cache size will get set to 0 when `type == SourceType::Annotations`. Not sure if that is problematic.

Reply:
When `type == SourceType::Annotations` cache is 0, because cache size is set only here and the default is 0.